contracts/smart-contract-wallet/SmartAccount.sol

```diff
@@ -288,13 +295,19 @@ contract SmartAccount is
         if (gasToken == address(0)) {
             // For ETH we will only adjust the gas price to not be higher than the actual used gas price
             payment = (gasUsed + baseGas) * (gasPrice < tx.gasprice ? gasPrice : tx.gasprice);
-            (bool success,) = receiver.call{value: payment}("");
-            require(success, "BSA011");
+            bool success;                                 
+            assembly {                                    
+                success := call(gas(), receiver, payment, 0, 0, 0, 0)
+            }
+            if(!success) revert TokenTransferFailed(address(0), receiver, payment);
         } else {
             payment = (gasUsed + baseGas) * (gasPrice) / (tokenGasPriceFactor);
-            require(transferToken(gasToken, receiver, payment), "BSA012");
+            if(!transferToken(gasToken, receiver, payment)) revert TokenTransferFailed(gasToken, receiver, payment);
```

Comment:
taking a note so it helps with resolving conflicts.
we should probably write test cases ti catch these error in foundry Test contracts and in off chain test scripts.